primitives/linux-port/column.cpp

```diff
@@ -1420,67 +1576,181 @@ void vectorizedFiltering(NewColRequestHeader* in, ColResultHeader* out, const T*
     }
   }
 
-  // main loop
-  // writeMask tells which values must get into the result. Includes values that matches filters. Can have
-  // NULLs. nonEmptyMask tells which vector coords are not EMPTY magics. nonNullMask tells which vector coords
-  // are not NULL magics.
-  for (uint16_t i = 0; i < iterNumber; ++i)
+  if (hasAuxCol)
+  {
+    using SimdTypeTemp = typename simd::IntegralToSIMD<uint8_t, KIND_UNSIGNED>::type;
+    using FilterTypeTemp = typename simd::StorageToFiltering<uint8_t, KIND_UNSIGNED>::type;
+    using VTAux = typename simd::SimdFilterProcessor<SimdTypeTemp, FilterTypeTemp>;
+    using SimdTypeAux = typename VTAux::SimdType;
+    using SimdWrapperTypeAux = typename VTAux::SimdWrapperType;
+    VTAux simdProcessorAux;
+    SimdTypeAux dataVecAux;
+    SimdTypeAux emptyFilterArgVecAux = simdProcessorAux.emptyNullLoadValue(emptyValueAux);
+    const uint8_t* origBlockAux = blockAux;
+    constexpr uint16_t VECTOR_SIZE_AUX = VT::vecByteSize;
+    uint16_t iterNumberAux = HAS_INPUT_RIDS ? ridSize / VECTOR_SIZE_AUX : srcSize / VECTOR_SIZE_AUX;
+    MT* nonEmptyMaskAux = (MT*) alloca(sizeof(MT) * iterNumberAux);
+    primitives::RIDType* origRidArray = ridArray;
+
+    for (uint16_t i = 0; i < iterNumberAux; ++i)
+    {
+      dataVecAux = simdDataLoad<VTAux, SimdWrapperTypeAux, HAS_INPUT_RIDS, uint8_t>(simdProcessorAux, blockAux,
+        origBlockAux, ridArray, i).v;
+      nonEmptyMaskAux[i] = simdProcessorAux.nullEmptyCmpNe(dataVecAux, emptyFilterArgVecAux);
+      blockAux += VECTOR_SIZE_AUX;
+      ridArray += VECTOR_SIZE_AUX;
+    }
+
+    ridArray = origRidArray;
+
+    MT (*getNonEmptyMaskPtr)(MT*, uint16_t);
+
+    switch(WIDTH)
+    {
+      case 1:
```

Comment:
We surely can do this w/o switch but using a mapping struct, e.g. getNonEmptyMaskPtr = getMask<WIDTH>().
template<int W>
consteval uint8_t* getMask() { return nullptr};
template<>
consteval uint8_t* getMask()<1> { return getNonEmptyMask1Byte; };
etc

Reply:
I have made this change, thanks for this feedback.